packages/gatsby-link/src/index.js

```diff
@@ -3,7 +3,7 @@ import { Link, NavLink } from "react-router-dom"
 import PropTypes from "prop-types"
 
 let pathPrefix = ``
-if (__PREFIX_PATHS__) {
+if (typeof __PREFIX_PATHS__ !== "undefined" && __PREFIX_PATHS__) {
```

Comment:
`undefined` is correct.

The `null` value represents an object with a non-existent reference. `undefined` is a clearer primitive, indicating that no value has been assigned at all.